### ISSUE TITLE
Florist potential rework suggestion

### ISSUE THREAD

**Issue by Shadizar** [ldtteam/minecolonies-features#764]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [x] I made sure that **this change request is not a duplicate of any existing change request.**

### Context 

- Minecraft Version:.1.19.2
- MineColonies Version:minecolonies-1.19.2-1.0.1371-ALPHA
- Related Mods and their Versions:

## Is your feature request related to a problem?

As a datapack editor I would like to be able to gate certain flowers behind a higher level florist, and to affect the rarity of some.

## Enhancement or Change description

Recipe-like, so that certain flowers could be level-gated in datapacks. For example, there are a handful of flowers in the Nether in some biome related mods. It would be good to be able to get them from the florist, but perhaps not til higher levels.
I would also recommend the ability to either weight them in relative frequency numerically, or via enumeration such as "common; uncommon; rare" or such.
This would allow datapack creators in modpacks to (for example) add wither rose as a very rare Level 5 product (some biome packs add wither roses even in overworld). Of course, vanilla minecolonies defaults would probably omit such and remain in line with current design, but the added flexibility would be great for datapack operations.

### Notes or related things

Comments from Mirality
Mirality — Today at 7:28 AM
Basically the way to do this would be to give it levelled loot tables, similar to the fisher and enchanter.  I like the idea, you should put it onto github.
(It wasn't practical to do this back when we were trying to automatically pick up all modded flowers, but since we switched to a custom tag due to Botania it's less weird to have custom loot tables.)

---
## Viewers

* Add a thumbs-up to the change request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.
